playwright/reporter.js

```diff
@@ -0,0 +1,83 @@
+const CI = require('../util/ci')
+const uploadTestResults = require('../util/uploadTestResults')
+
+/**
+ * JSDoc Imports
+ *
+ * @typedef {import('@playwright/test/reporter').FullConfig} FullConfig
+ * @typedef {import('@playwright/test/reporter').FullResult} FullResult
+ * @typedef {import('@playwright/test/reporter').Reporter} Reporter
+ * @typedef {import('@playwright/test/reporter').Suite} Suite
+ * @typedef {import('@playwright/test/reporter').TestCase} TestCase
+ * @typedef {import('@playwright/test/reporter').TestResult} TestResult
+ */
+
+/**
+ * A playwright reporter that uploads test results to Buildkite Test Analytics
+ *
+ * @typedef {Reporter}
+ */
+class PlaywrightBuildkiteAnalyticsReporter {
+
+  constructor() {
+    this._testResults = []
+    this._testEnv = (new CI()).env();
+  }
+  
+  onBegin() {}
+
+  onEnd() {
+    uploadTestResults(this._testEnv, this._testResults)
+  }
+
+  onTestBegin() {}
+
+  /**
+   *
+   * @param {TestCase} test
+   * @param {TestResult} testResult
+   */
+  onTestEnd(test, testResult) {
+    this._testResults.push({
+      'id': test.id,
+      'scope': test.titlePath.join(' '),
+      'name': test.title,
+      'location': test.location,
+      'file_name': test.location,
+      'result': this.analyticsResult(testResult.status),
+      'failure_reason': testResult.error,
```

Comment:
`failure_reason` expects a `string`. You can see the documentation [here](https://buildkite.com/docs/test-analytics/importing-json#json-test-results-data-reference-test-result-objects).
```suggestion
      'failure_reason': testResult.error.message,
```